contracts/multisig-prover/src/payload.rs

```diff
@@ -70,6 +80,21 @@ impl Payload {
                 abi::execute_data::encode(verifier_set, signers_with_sigs, &payload_hash, payload)
             }
             Encoder::Bcs => todo!(),
+            Encoder::Solana => {
+                let enc_signatures = signers_with_sigs
+                    .iter()
+                    .map(to_weighted_signature)
+                    .collect_vec();
+
+                let bytes = axelar_encoding::encode::<1024>( // Todo reason about this "1024" magic number.
+                    &to_worker_set(&verifier_set),
+                    enc_signatures,
+                    axelar_encoding::types::Payload::from(payload),
+                )
+                .map_err(|e| ContractError::SolEncodingError(e.to_string()))?;
+
+                Ok(HexBinary::from(bytes)) // Todo, what kind of conversion if any comes from here. Can we expect hex repr directly ?
```

Comment:
Similar to other comments that deals with raw data. We need to ensure this is the correct representation.

Reply:
So if I understand this correctly: we encode the bytes with Rkyv, and then we hex-encode it (because the multisig-prover expects it to be like that). If I understand correctly, the payload hash has already been signed beforehand, right?

Reply:
Yeah, i think you are right. Nothing to do here really. The caller code just knows an rkyv encoded data is returned in HexBinary, end of history. Fixed at 1766b3b2b52f4f9b3d0d40b7847629b78a02b760